draft-ietf-quic-qlog-quic-events.md

```diff
@@ -1014,19 +1014,42 @@ important, the packet_received event can be used instead.
 
 In some implementations, it can be difficult to log frames directly, even
 when using packet_sent and packet_received events. For these cases, this event
-also contains the direct packet_number field, which can be used to more explicitly
-link this event to the packet_sent/received events.
+also contains the packet_number field, which can be used to more explicitly
+link this event to the packet_sent/received events. The field is an array, which
+supports using a single "frames_processed" event for multiple frames received
+over multiple packets. To map between frames and packets, the position and order
+of entries in the "frames" and "packet_number" is used. If the "packet_number"
+field is used, each frame MUST have a corresponding packet number at the same
+index.
 
 Definition:
 
 ~~~ cddl
 QUICFramesProcessed = {
     frames: [* $QuicFrame]
-    ? packet_number: uint64
+    ? packet_number: [* uint64]
```

Comment:
```suggestion
    ? packet_numbers: [* uint64]
```

For consistency with existing events like `packets_acked` and partially the `datagram_*` events that talk about `datagram_ids` not `datagram_id`